include/pinocchio/multibody/joint/joint-ellipsoid.hpp

```diff
@@ -0,0 +1,834 @@
+//
+// Copyright (c) 2025 INRIA
+//
+
+#ifndef __pinocchio_multibody_joint_ellipsoid_hpp__
+#define __pinocchio_multibody_joint_ellipsoid_hpp__
+
+#include "pinocchio/macros.hpp"
+#include "pinocchio/multibody/joint/joint-base.hpp"
+#include "pinocchio/multibody/joint-motion-subspace.hpp"
+#include "pinocchio/math/sincos.hpp"
+#include "pinocchio/math/matrix.hpp"
+
+namespace pinocchio
+{
+  template<typename Scalar, int Options>
+  struct JointEllipsoidTpl;
+
+  template<typename Scalar, int Options>
+  struct JointMotionSubspaceEllipsoidTpl;
+
+  template<typename Scalar, int Options>
+  struct SE3GroupAction<JointMotionSubspaceEllipsoidTpl<Scalar, Options>>
+  {
+    typedef Eigen::Matrix<Scalar, 6, 3, Options> ReturnType;
+  };
+
+  template<typename Scalar, int Options, typename MotionDerived>
+  struct MotionAlgebraAction<JointMotionSubspaceEllipsoidTpl<Scalar, Options>, MotionDerived>
+  {
+    typedef Eigen::Matrix<Scalar, 6, 3, Options> ReturnType;
+  };
+
+  template<typename Scalar, int Options, typename ForceDerived>
+  struct ConstraintForceOp<JointMotionSubspaceEllipsoidTpl<Scalar, Options>, ForceDerived>
+  {
+    typedef Eigen::Matrix<Scalar, 3, 1, Options> ReturnType;
+  };
+
+  template<typename Scalar, int Options, typename ForceSet>
+  struct ConstraintForceSetOp<JointMotionSubspaceEllipsoidTpl<Scalar, Options>, ForceSet>
+  {
+    typedef Eigen::Matrix<Scalar, 3, Eigen::Dynamic, Options> ReturnType;
```

Comment:
This will do a dynamic allocation. You should use the standard definition:

```cpp
    typedef
      typename traits<JointMotionSubspaceEllipsoidTpl<Scalar, Options>>::DenseBase DenseBase;
    typedef
      typename MatrixMatrixProduct<Eigen::Transpose<const DenseBase>, ForceSet>::type ReturnType;
```